client.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"net/url"
+
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+)
+
+var (
+	tag = "GPTeaching"
```

Comment:
we are trying to create a generalized client, so we shouldn't have anything hard coded like this. Further, I would expect the caller to pass in a tag when a function is called, so shouldn't have a var here for that.

comment about general structure of the client: look at something like https://github.com/kothar/go-backblaze
--> package name should be "iconik"
--> you should have a struct which passes in the config for the client (i.e. the appID and token), and then you should have a constructor called NewClient or InitClient, and it returns a pointer to an IconikClient object
--> you have a call (exported) like SearchWithTags(tags []string) which returns a SearchResponse object

So there shouldn't be a main package, and we certainly shouldn't have any of our keys in the repo, and it should be clear what is exported and what isn't.

Reply:
Done!